src/tokens/VestingContract.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "../interfaces/IVestingContract.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title VestingContract - To be deployed on ETH mainnet
+ * Created following OpenZeppelin Contracts (last updated v5.0.0) (finance/VestingWallet.sol)
+ *
+ * @dev A vesting wallet is an ownable contract that can receive  ERC-20 tokens, and release these
+ * assets to "beneficiary", according to a vesting schedule and a mandatory one-year lock.
+ *
+ * Any assets transferred to this contract will follow the vesting schedule as if they were locked from the beginning.
+ * Consequently, if the vesting has already started, any amount of tokens sent to this contract will (at least partly)
+ * be immediately releasable.
+ *
+ * By setting the duration to 0, one can configure this contract to behave like an asset timelock that hold tokens for
+ * a beneficiary until a specified time + lock time.
+ *
+ */
+contract VestingContract is Ownable, IVestingContract {
+    event ERC20Released(address indexed beneficiary, uint256 amount);
+
+    address public immutable override kintoToken;
+    uint256 public constant override LOCK_PERIOD = 365 days;
+
+    mapping(address => uint256) private _erc20Released;
+    mapping(address => uint256) private _grant;
+    mapping(address => uint256) private _start;
+    mapping(address => uint256) private _duration;
+
+    uint256 public override totalAllocated;
+    uint256 public override totalReleased;
+
+    /**
+     * @dev Sets the sender as the initial owner, the beneficiary as the pending owner, the start timestamp and the
+     * vesting duration of the vesting wallet.
+     */
+    constructor(address token) Ownable() {
```

Comment:
On the latest version of OZ we will need to update this to receive the owner: `Ownable(msg.sender)`